src/core/Akka/Actor/Inbox.Actor.cs

```diff
@@ -185,29 +185,21 @@ protected override bool Receive(object message)
 
             if (_clients.Count == 0)
             {
-                if (_currentDeadline != null)
-                {
-                    _currentDeadline.Item2.Cancel();
-                    _currentDeadline = null;
-                }
+                _currentDeadline.Item2.Cancel();
```

Comment:
While `_currentDeadline` can't be null anymore `Item2` can, so we should still check if it's not null before cancelling